packages/@n8n/nodes-langchain/nodes/tools/ToolHttpRequest/utils.ts

```diff
@@ -0,0 +1,793 @@
+				return await ctx.helpers.httpRequest(options);
+			};
+		} else if (genericType === 'httpHeaderAuth') {
+			const headerAuth = await ctx.getCredentials('httpHeaderAuth', itemIndex);
+
+			return async (options: IHttpRequestOptions) => {
+				options.headers![headerAuth.name as string] = headerAuth.value;
+				return await ctx.helpers.httpRequest(options);
+			};
+		} else if (genericType === 'httpQueryAuth') {
+			const queryAuth = await ctx.getCredentials('httpQueryAuth', itemIndex);
+
+			return async (options: IHttpRequestOptions) => {
+				if (!options.qs) {
+					options.qs = {};
+				}
+				options.qs[queryAuth.name as string] = queryAuth.value;
+				return await ctx.helpers.httpRequest(options);
+			};
+		} else if (genericType === 'httpCustomAuth') {
+			const customAuth = await ctx.getCredentials('httpCustomAuth', itemIndex);
+
+			return async (options: IHttpRequestOptions) => {
+				const auth = jsonParse<IRequestOptionsSimplified>((customAuth.json as string) || '{}', {
+					errorMessage: 'Invalid Custom Auth JSON',
+				});
+				if (auth.headers) {
+					options.headers = { ...options.headers, ...auth.headers };
+				}
+				if (auth.body) {
+					options.body = { ...(options.body as IDataObject), ...auth.body };
+				}
+				if (auth.qs) {
+					options.qs = { ...options.qs, ...auth.qs };
+				}
+				return await ctx.helpers.httpRequest(options);
+			};
+		} else if (genericType === 'oAuth1Api') {
+			return async (options: IHttpRequestOptions) => {
+				return await ctx.helpers.requestOAuth1.call(ctx, 'oAuth1Api', options);
+			};
+		} else if (genericType === 'oAuth2Api') {
+			return async (options: IHttpRequestOptions) => {
+				return await ctx.helpers.requestOAuth2.call(ctx, 'oAuth1Api', options, {
+					tokenType: 'Bearer',
+				});
+			};
+		}
+	} else if (credentialsType === 'predefinedCredentialType') {
+		const predefinedType = ctx.getNodeParameter('nodeCredentialType', itemIndex) as string;
+		const additionalOptions = getOAuth2AdditionalParameters(predefinedType);
+
+		return async (options: IHttpRequestOptions) => {
+			return await ctx.helpers.requestWithAuthentication.call(
+				ctx,
+				predefinedType,
+				options,
+				additionalOptions && { oauth2: additionalOptions },
+				itemIndex,
+			);
+		};
+	}
+
+	return async (options: IHttpRequestOptions) => {
+		return await ctx.helpers.httpRequest(options);
+	};
+};
+
+export const configureResponseOptimizer = (ctx: IExecuteFunctions, itemIndex: number) => {
+	const optimizeResponse = ctx.getNodeParameter('optimizeResponse', itemIndex, false) as boolean;
+
+	if (optimizeResponse) {
+		const responseType = ctx.getNodeParameter('responseType', itemIndex) as
+			| 'json'
+			| 'text'
+			| 'html';
+
+		let maxLength = 0;
+		const truncateResponse = ctx.getNodeParameter('truncateResponse', itemIndex, false) as boolean;
+
+		if (truncateResponse) {
+			maxLength = ctx.getNodeParameter('maxLength', itemIndex, 0) as number;
+		}
+
+		if (responseType === 'html') {
```

Comment:
This function is quite large; perhaps we could split out the responseType parsers into separate methods